cmd/argo/commands/root.go

```diff
@@ -133,5 +138,20 @@ If your server is behind an ingress with a path (you'll be running "argo server
 	command.PersistentFlags().IntVar(&glogLevel, "gloglevel", 0, "Set the glog logging level")
 	command.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false, "Enabled verbose logging, i.e. --loglevel debug")
 
+	viper.AutomaticEnv()
+	viper.SetEnvPrefix("ARGO")
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_", ".", "_"))
+	if err := viper.BindPFlags(command.PersistentFlags()); err != nil {
+		log.Fatal(err)
+	}
+	command.PersistentFlags().VisitAll(func(f *pflag.Flag) {
+		if !f.Changed && viper.IsSet(f.Name) {
+			val := viper.Get(f.Name)
+			if err := command.PersistentFlags().Set(f.Name, fmt.Sprintf("%v", val)); err != nil {
+				log.Fatal(err)
+			}
+		}
+	})
+
```

Comment:
can we add comments to make it clear what all of this is doing?

Reply:
it's not commented in [other files](https://github.com/argoproj/argo-workflows/blob/a47e46208cd0029ef2e35019207d0b1ee4f85ea4/cmd/argo/commands/server.go#L240), but I could add explanatory comments to all of the files

Reply:
that would be great

Reply:
added comments in the most recent [commit](https://github.com/argoproj/argo-workflows/pull/12145/commits/31793b029a0676706713c7ffaa7a01008822d738)